File: AssetManager/UserInterface/Forms/SplashScreenForm.vb
Imports System.Deployment.Application

Public Class SplashScreenForm

    Private Sub SplashScreen1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        Me.FormBorderStyle = FormBorderStyle.None
        Using p As New Drawing2D.GraphicsPath()
            p.StartFigure()
            p.AddArc(New Rectangle(0, 0, 40, 40), 180, 90)
            p.AddLine(40, 0, Me.Width - 40, 0)
            p.AddArc(New Rectangle(Me.Width - 40, 0, 40, 40), -90, 90)
            p.AddLine(Me.Width, 40, Me.Width, Me.Height - 40)
            p.AddArc(New Rectangle(Me.Width - 40, Me.Height - 40, 40, 40), 0, 90)
            p.AddLine(Me.Width - 40, Me.Height, 40, Me.Height)
            p.AddArc(New Rectangle(0, Me.Height - 40, 40, 40), 90, 90)
            p.CloseFigure()
            Me.Region = New Region(p)
        End Using
        If ApplicationDeployment.IsNetworkDeployed Then
            Version.Text = ApplicationDeployment.CurrentDeployment.CurrentVersion.ToString
        Else
            Version.Text = "Debug"
        End If
        Copyright.Text = My.Application.Info.Copyright
    End Sub

    Overloads Sub Hide()
        Me.Dispose()
    End Sub

    Public Sub SetStatus(Text As String)
        lblStatus.Text = Text
        Me.Refresh()
    End Sub

End Class

File: AssetManager/UserInterface/Forms/SplashScreenForm.cs
using AssetManager.UserInterface.CustomControls;
using System.Deployment.Application;
using System.Drawing;
using System.Reflection;
using System.Windows.Forms;

namespace AssetManager.UserInterface.Forms
{
    public partial class SplashScreenForm : ExtendedForm
    {
        private void SplashScreen1_Load(object sender, System.EventArgs e)
        {
            this.FormBorderStyle = FormBorderStyle.None;
            using (System.Drawing.Drawing2D.GraphicsPath p = new System.Drawing.Drawing2D.GraphicsPath())
            {
                p.StartFigure();
                p.AddArc(new Rectangle(0, 0, 40, 40), 180, 90);
                p.AddLine(40, 0, this.Width - 40, 0);
                p.AddArc(new Rectangle(this.Width - 40, 0, 40, 40), -90, 90);
                p.AddLine(this.Width, 40, this.Width, this.Height - 40);
                p.AddArc(new Rectangle(this.Width - 40, this.Height - 40, 40, 40), 0, 90);
                p.AddLine(this.Width - 40, this.Height, 40, this.Height);
                p.AddArc(new Rectangle(0, this.Height - 40, 40, 40), 90, 90);
                p.CloseFigure();
                this.Region = new Region(p);
            }
            if (ApplicationDeployment.IsNetworkDeployed)
            {
                Version.Text = ApplicationDeployment.CurrentDeployment.CurrentVersion.ToString();
            }
            else
            {
                Version.Text = "Debug";
            }
            Assembly asm = Assembly.GetExecutingAssembly();
            string copyright = ((AssemblyCopyrightAttribute)asm.GetCustomAttribute(typeof(AssemblyCopyrightAttribute))).Copyright;
            Copyright.Text = copyright;
        }

        public void SetStatus(string text)
        {
            lblStatus.Text = text;
            this.Refresh();
        }

        public SplashScreenForm()
        {
            Load += SplashScreen1_Load;
            InitializeComponent();
        }
    }
}